Model/Reclaim.php

```diff
@@ -47,10 +75,114 @@ public function __construct(
      * @api
      * @return string
      */
-    public function reclaim(){
+    public function reclaim()
+    {
         return $this->_klaviyoHelper->getVersion();
     }
 
+    /**
+     * Returns the Klaviyo log file
+     * 
+     * @api
+     * @return string
+     */
+    public function getLog()
+    {
+        $log = file($this->_dir->getPath('log') . '/klaviyo.log');
+        if ($log != "") {
+            return $log;
+        } else {
+            return array (
+                'message' => 'Unable to retrieve log file'
+            );
+        }
+    }
+
+    /**
+     * Cleans the Klaviyo log file
+     * 
+     * @api
+     * @param string $date
+     * @return boolean
+     */
+    public function cleanLog($date)
+    {
+        //attempt to parse unix timestamp from api request parameter
+        $cursor = strtotime($date);
+
+        //check if we were able to parse the timestamp
+        //if no timestamp, return failure message
+        if ($cursor == "")
+        {
+            $response = array(
+                'message' => 'Unable to parse timestamp: ' . $date
+            );
+            $this->klaviyoLogger->info("cleanLog failed: unable to parse timestamp from: " . $date);
+            return $response;
+        }
+        
+        //get log file path and do the old switcheroo in preparation for cleaning
+        $path = $this->_dir->getPath('log') . '/klaviyo.log';
+        $old = $path . ".old";
+        rename($path, $old);
+
+        //open file streams
+        $input = fopen($old, "rb");
```

Comment:
so if i get this right the logic is:

1. when we go to clean the log, we make a copy
2. we create a new log file, then for everything in the old file that the timestamp is > the $date we will write to the new file
3. purge the old file?

Reply:
correct, should we be doing something different?